cast-receiver/src/status_ui_handler.ts

```diff
@@ -1,104 +1,102 @@
-import { PlayerStatusEventType } from "./types";
 import { Icons } from "./library";
 
 /**
  * StatusUIHandler is responsible for managing status UI of the application.
- * It implements a callback for subscribing to the PlayerManager events (binding
- * in index.ts) and updates the div container (passed as a constructor argument).
+ * It manages the div container passed via the constructor argument.
  */
 export default class StatusUIHandler {
-  private static readonly READY_TO_CAST_MSG = "Ready to cast";
+  static readonly IDLE_STATUS_ID = "idle";
+  static readonly CASTING_STATUS_ID = "casting";
+  static readonly LOADER_STATUS_ID = "loader";
 
-  static readonly READY_TO_CAST_ID = "ready-to-cast";
-  static readonly LOADING_OPACITY = 0.5;
-  static readonly PAUSED_OPACITY = 0.65;
-  static readonly NORMAL_OPACITY = 0.9;
+  private static readonly IDLE_STATUS_ICON = "cast";
+  private static readonly CASTING_STATUS_ICON = "cast";
+  private static readonly LOADER_STATUS_ICON = "loader";
+
+  private static readonly IDLE_STATUS_MSG = "Ready to cast";
+  private static readonly CASTING_STATUS_MSG = "A device is casting";
 
   private container: HTMLDivElement;
-  private readyToCastElement: HTMLSpanElement;
+  private statusElements: Map<string, HTMLElement> = new Map();
 
   constructor(statusContainer: HTMLDivElement) {
     this.container = statusContainer;
-    this.readyToCastElement = this.createReadyToCastElement();
+    this.statusElements.set(
+      StatusUIHandler.IDLE_STATUS_ID,
+      this.createStatusElement(
+        StatusUIHandler.IDLE_STATUS_ID,
+        StatusUIHandler.IDLE_STATUS_ICON,
+        StatusUIHandler.IDLE_STATUS_MSG
+      )
+    );
+
+    this.statusElements.set(
+      StatusUIHandler.CASTING_STATUS_ID,
+      this.createStatusElement(
+        StatusUIHandler.CASTING_STATUS_ID,
+        StatusUIHandler.CASTING_STATUS_ICON,
+        StatusUIHandler.CASTING_STATUS_MSG
+      )
+    );
+
+    this.statusElements.set(
+      StatusUIHandler.LOADER_STATUS_ID,
+      this.createStatusElement(
+        StatusUIHandler.LOADER_STATUS_ID,
+        StatusUIHandler.LOADER_STATUS_ICON,
+        ""
+      )
+    );
   }
 
   private createIcon(id: string): HTMLElement {
     const template = document.createElement("template");
-    template.innerHTML = `<svg id="${id}" class="sound"><use xlink:href="${Icons[id]}" /><svg>`;
+    template.innerHTML = `<svg id="${id}" class="icon"><use xlink:href="${Icons[id]}" /><svg>`;
```

Comment:
Codacy found an issue: [Unsafe assignment to innerHTML](https://app.codacy.com/manual/ashutoshgngwr/noice/pullRequest?prid=6042010)